lib/jsonapi/fetchdata/parameters/parsers/sort.rb

```diff
@@ -8,7 +8,11 @@ class Sort < Parser
 
           def parse params
             if params && (params.size > 0)
-              [params].flatten.map{ |col| col.scan(/[-\w]+/)}.flatten.map { |c| (c[0] == '-') ? Hash[c[1..-1], 'desc'] : c }
+              if Rails::VERSION::MAJOR >= 4
+                [params].flatten.map{ |col| col.scan(/[-\w]+/)}.flatten.map { |c| (c[0] == '-') ? Hash[c[1..-1], 'desc'] : c }
```

Comment:
For rails 3 the hash format doesn't work well that is why we need to fallback to the string version